panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6b8aadc]

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.8.2

### AzureRM Provider Version

3.102

### Affected Resource(s)/Data Source(s)

azurerm_key_vault_access_policy, azurerm_web_pubsub_network_acl

### Terraform Configuration Files

```hcl
My module is pretty complex. I will try to reproduce it with a similar one...
```


### Debug Output/Panic Output

```shell
`
│ Error: Plugin did not respond
│ 
│   with module.webpubsub.module.keyvault_accesspolicy_wps[0].azurerm_key_vault_access_policy.key_vault_access_policies,
│   on .terraform/modules/webpubsub.keyvault_accesspolicy_wps/resource-akv-accesspolicy.tf line 1, in resource "azurerm_key_vault_access_policy" "key_vault_access_policies":
│    1: resource "azurerm_key_vault_access_policy" "key_vault_access_policies" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.webpubsub.azurerm_web_pubsub_network_acl.webpubsub[0],
│   on .terraform/modules/webpubsub/main.tf line 76, in resource "azurerm_web_pubsub_network_acl" "webpubsub":
│   76: resource "azurerm_web_pubsub_network_acl" "webpubsub" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.webpubsub.azurerm_web_pubsub_custom_certificate.custom_domain[0],
│   on .terraform/modules/webpubsub/resources-dns.tf line 1, in resource "azurerm_web_pubsub_custom_certificate" "custom_domain":
│    1: resource "azurerm_web_pubsub_custom_certificate" "custom_domain" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-azurerm_v3.101.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6b8aadc]

goroutine 352 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/signalr.resourceWebPubsubNetworkACLCreateUpdate(0xc001a3c180, {0x76d02e0?, 0xc0015a5200})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/signalr/web_pubsub_network_acl_resource.go:168 +0x55c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x8eee870?, {0x8eee870?, 0xc00270ffb0?}, 0xd?, {0x76d02e0?, 0xc0015a5200?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:766 +0x163

`
```


### Expected Behaviour

It should assing the KeyVault RBAC and setup WPS ACL. 

### Actual Behaviour

It crashes with the above error.

### Steps to Reproduce

terraform apply

### Important Factoids

_No response_

### References

_No response_

I got the same issue with the azurerm provider version 3.102.0. If I use the version 3.101.0 it is working. 
Here are the errors I received, when running a terraform plan.

--< snip  >--
Stack trace from the terraform-provider-azurerm_v3.102.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6268dcc]

goroutine 4987 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/lighthouse.resourceLighthouseDefinitionRead(0xc001f07200, {0x75ebc20?, 0xc001be9200?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/lighthouse/lighthouse_definition_resource.go:273 +0x6c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x8dec7b0?, {0x8dec7b0?, 0xc00217f170?}, 0xd?, {0x75ebc20?, 0xc001be9200?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:783 +0x163
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0017b4fc0, {0x8dec7b0, 0xc00217f170}, 0xc0026596c0, {0x75ebc20, 0xc001be9200})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:1089 +0x552
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0008e5e30, {0x8dec7b0?, 0xc00217ef60?}, 0xc0055a3880)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:649 +0x48a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc001667360, {0x8dec7b0?, 0xc0026fb9b0?}, 0xc00483a780)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:789 +0x48b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x8229260?, 0xc001667360}, {0x8dec7b0, 0xc0026fb9b0}, 0xc0019db420, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:431 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004d0000, {0x8e15580, 0xc0019b0680}, 0xc003616360, 0xc001c306f0, 0xe641de8, 0x0)
	google.golang.org/grpc@v1.58.3/server.go:1374 +0xde7
google.golang.org/grpc.(*Server).handleStream(0xc0004d0000, {0x8e15580, 0xc0019b0680}, 0xc003616360, 0x0)
	google.golang.org/grpc@v1.58.3/server.go:1751 +0x9e7
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.58.3/server.go:986 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 16
	google.golang.org/grpc@v1.58.3/server.go:997 +0x145

Error: The terraform-provider-azurerm_v3.102.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
--< snap >--

--- a/internal/services/signalr/web_pubsub_network_acl_resource.go
+++ b/internal/services/signalr/web_pubsub_network_acl_resource.go
@@ -165,7 +165,7 @@ func resourceWebPubsubNetworkACLCreateUpdate(d *pluginsdk.ResourceData, meta int
 
 	if defaultAction == webpubsub.ACLActionAllow && networkACL.PublicNetwork.Allow != nil && len(*networkACL.PublicNetwork.Allow) != 0 {
 		return fmt.Errorf("when `default_action` is `Allow` for `public_network`, `allowed_request_types` cannot be specified")
-	} else if defaultAction == webpubsub.ACLActionDeny && len(*networkACL.PublicNetwork.Deny) != 0 {
+	} else if defaultAction == webpubsub.ACLActionDeny && networkACL.PublicNetwork.Deny != nil && len(*networkACL.PublicNetwork.Deny) != 0 {
 		return fmt.Errorf("when `default_action` is `Deny` for `public_network`, `denied_request_types` cannot be specified")
 	}
 

